gradio/interface.py

```diff
@@ -447,10 +454,82 @@ def clean_html(raw_html):
         if self.analytics_enabled:
             utils.initiated_analytics(data)
 
-        # Alert user if a more recent version of the library exists
         utils.version_check()
         Interface.instances.add(self)
 
+        param_names = inspect.getfullargspec(self.predict[0])[0]
+        for component, param_name in zip(self.input_components, param_names):
+            if component.label is None:
+                component.label = param_name
+        for i, component in enumerate(self.output_components):
+            if component.label is None:
+                component.label = "output_" + str(i)
+
+        if self.allow_flagging != "never":
+            self.flagging_callback.setup(
+                self.input_components + self.output_components, self.flagging_dir
+            )
+
+        self.blocks = Blocks()
+        with self.blocks:
+            if self.title:
+                Markdown(
+                    "<h1 style='text-align: center; margin-bottom: 1rem'>"
+                    + self.title
+                    + "</h1>"
+                )
+            if self.description:
+                Markdown(self.description)
+            with Row():
+                with Column(
+                    css={
+                        "background-color": "rgb(249,250,251)",
+                        "padding": "0.5rem",
+                        "border-radius": "0.5rem",
+                    }
+                ):
+                    for component in self.input_components:
+                        Block.__init__(component)
```

Comment:
I think this is necessary for the components to show up.

Perhaps there should be a better way to do this, though? For example, components could have a `.render()` method that adds them to the Block? So you could just replace this line with `component.render()`

Reply:
Yeah it's necessary because normally when Blocks are initialized, they are automatically added as children to context they are created in. But since all Interface components are created before hand, they need to be re-initialized inside the Block context.

Reply:
What do you think about the render() suggestion? It’s more readable code imo

Reply:
Done.